### PR TITLE
docs: add installation page

### PR DESCRIPTION
**Pull request by matklad**
- The official way to install TigerBeetle is to download a release binary.
- Don't tether us to GitHub releases, provide a table with downloads and treat URLs as opaque
- Don't advertise `-debug` artifacts, we want to keep the release page lean.
- Don't mark releases as `--prerelease` --- this is required for `/latest` links to work.

Future work: don't `.zip` our executable, it should be small enough as is.

### CODE REVIEW

**Comment by matklad** [comment]
image

**Comment by jorangreef** [comment]
Glorious.

File: docs/installation.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+sidebar_position: 2
+---
+
+# Installation
+
+TigerBeetle is a single small statically linked binary, download it here:
+
+|               | Linux                            | Windows                          | MacOs                             |
```
Comment (jorangreef) [inline docs/installation.md]:
```suggestion
|               | Linux                            | Windows                          | macOS                             |
```

File: docs/installation.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+sidebar_position: 2
+---
+
+# Installation
+
+TigerBeetle is a single small statically linked binary, download it here:
```
Comment (jorangreef) [inline docs/installation.md]:
```suggestion
TigerBeetle is a single, small, statically linked binary, download it here:
```

File: docs/installation.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+| :------------ | :------------------------------- | :------------------------------- | :-------------------------------- |
+| x86_64        |[tigerbeetle-x86_64-linux.zip]    |[tigerbeetle-x86_64-windows.zip]  | [tigerbeetle-universal-macos.zip] |
+| aarch64       |[tigerbeetle-aarch64-linux.zip]   |             N/A                  | [tigerbeetle-universal-macos.zip] |
+
+[tigerbeetle-aarch64-linux.zip]: https://github.com/tigerbeetle/tigerbeetle/releases/latest/download/tigerbeetle-aarch64-linux.zip
+[tigerbeetle-universal-macos.zip]: https://github.com/tigerbeetle/tigerbeetle/releases/latest/download/tigerbeetle-universal-macos.zip
+[tigerbeetle-x86_64-linux.zip]: https://github.com/tigerbeetle/tigerbeetle/releases/latest/download/tigerbeetle-x86_64-linux.zip
+[tigerbeetle-x86_64-windows.zip]: https://github.com/tigerbeetle/tigerbeetle/releases/latest/download/tigerbeetle-x86_64-windows.zip
+
+While not required, a Docker container is also available:
```
Comment (jorangreef) [inline docs/installation.md]:
```suggestion
While not required (TigerBeetle is designed to be easy to deploy as a single binary), a Docker container is also available:
```

Comment (jorangreef) [reply docs/installation.md]: Adding a little redundancy here on purpose to communicate the "single binary" experience idea and make it obvious.

File: docs/installation.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+<https://github.com/tigerbeetle/tigerbeetle/pkgs/container/tigerbeetle>
+
+
+Client libraries for .Net, Go, Java, and NodeJS are available in respective registries:
```
Comment (jorangreef) [inline docs/installation.md]:
```suggestion
Client libraries for .Net, Go, Java, and NodeJS are available in the respective registries:
```

File: docs/installation.md (hunk 3 above)
Comment (matklad) [reply docs/installation.md]: :100:

**Comment by cb22** [comment]
Think this needs a warning or note about versioning - like we have on our release notes?

File: docs/installation.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+[tigerbeetle-x86_64-windows.zip]: https://github.com/tigerbeetle/tigerbeetle/releases/latest/download/tigerbeetle-x86_64-windows.zip
+
+While not required (TigerBeetle is designed to be easy to deploy as a single binary), a Docker
+container is also available:
+
+<https://github.com/tigerbeetle/tigerbeetle/pkgs/container/tigerbeetle>
+
+
+Client libraries for .Net, Go, Java, and NodeJS are available in the respective registries:
+
```
Comment (sentientwaffle) [inline docs/installation.md:26]: I think this is what Fed [pointed out](https://github.com/tigerbeetle/tigerbeetle/pull/1321#issuecomment-1827378809), but we should mention that the server/client versions should match, e.g.

> NOTE: You must run the same version of server and client. We do
> not yet follow semantic versioning where all patch releases are
> interchangeable.

Comment (matklad) [reply docs/installation.md:26]: And that we are also non-production ready yet!

**Comment by matklad** [comment]
We probably should integrate link to this page into our quickstart examples & readme, but I want to do that separately

**Review by sentientwaffle — APPROVED** [review]

**Pull request merged**